### PR TITLE
Define CentOS vs RHEL dependencies

### PR DESCRIPTION
**Pull request by alfredodeza**
And install them accordingly. This should help in situations where RHEL systems will not have the same packages as a CentOS box (or shouldn't like EPEL on RHEL).

Should fix issue #584

### CODE REVIEW

**Comment by ktdreyer** [comment]
LGTM

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 1]
```diff
@@ -4,7 +4,14 @@
     name: "{{ item }}"
     state: present
   with_items: redhat_package_dependencies
-  when: ansible_pkg_mgr == "yum"
```
Comment (andrewschoen) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: You probably still need this condition as we're supporting ``dnf`` below as well.

Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: adding `and ansible_pkg_mgr == "yum"` is probably safier.

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 2]
```diff
@@ -4,7 +4,14 @@
...
+  when: ansible_distribution == "RedHat"
+
+- name: install dependencies
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: centos_package_dependencies
+  when: ansible_distribution == "CentOS"
 
 - name: install dependencies
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: and `ansible_distribution == "RedHat"`?

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 3]
```diff
@@ -4,7 +4,14 @@
...
   with_items: redhat_package_dependencies
-  when: ansible_pkg_mgr == "yum"
+  when: ansible_distribution == "RedHat"
+
+- name: install dependencies
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: centos_package_dependencies
+  when: ansible_distribution == "CentOS"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: can you add a condition for `dnf` as well for CentOS?

Comment (alfredodeza) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: yep! good catch

File: roles/ceph-common/tasks/installs/install_on_redhat.yml (hunk 2 above)
Comment (alfredodeza) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: sure

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 4]
```diff
@@ -4,14 +4,21 @@
     name: "{{ item }}"
     state: present
   with_items: redhat_package_dependencies
-  when: ansible_pkg_mgr == "yum"
+  when: ansible_distribution == "RedHat" and ansible_pkg_mgr == "yum"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: please use the following syntax style:

```
when:
  ansible_distribution == "RedHat" and
  ansible_pkg_mgr == "yum"
```

Easier to read :)

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 5]
```diff
@@ -4,14 +4,21 @@
...
+
+- name: install dependencies
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: centos_package_dependencies
+  when: ansible_distribution == "CentOS" and ansible_pkg_mgr == "yum"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: dito

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 6]
```diff
@@ -4,14 +4,21 @@
...
 
 - name: install dependencies
   dnf:
     name: "{{ item }}"
     state: present
   with_items: redhat_package_dependencies
-  when: ansible_pkg_mgr == "dnf"
+  when: ansible_distribution == "CentOS" and ansible_pkg_mgr == "dnf"
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: dito

**Comment by leseb** [comment]
Please rebase :)

**Comment by alfredodeza** [comment]
ouch, that rebase is going to take a while, everything changed in that file

**Comment by alfredodeza** [comment]
test this please

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 7]
```diff
@@ -4,14 +4,24 @@
     name: "{{ item }}"
     state: present
   with_items: redhat_package_dependencies
-  when: ansible_pkg_mgr == "yum"
+  when: ansible_distribution == "RedHat" and
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: sorry to be picky but let's keep things consistent :), so

```
when:
  ansible_distribution == "RedHat" and
  ansible_pkg_mgr == "yum"
```

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 8]
```diff
@@ -4,14 +4,24 @@
...
+        ansible_pkg_mgr == "yum"
+
+- name: install dependencies
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: centos_package_dependencies
+  when: ansible_distribution == "CentOS" and
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: dito

File: roles/ceph-common/tasks/installs/install_on_redhat.yml [hunk 9]
```diff
@@ -4,14 +4,24 @@
...
+        ansible_pkg_mgr == "yum"
 
 - name: install dependencies
   dnf:
     name: "{{ item }}"
     state: present
   with_items: redhat_package_dependencies
-  when: ansible_pkg_mgr == "dnf"
+  when: ansible_distribution == "CentOS" and
```
Comment (leseb) [inline roles/ceph-common/tasks/installs/install_on_redhat.yml]: dito

**Comment by leseb** [comment]
Crap something went wrong with one of my last commit.
Do you mind rebasing again?

Sorry for the pain...

**Comment by alfredodeza** [comment]
@leseb as long as you promise to merge this before you merge anything else because I swear I have crocodile tears coming out of my scalp trying to rebase this

**Comment by ktdreyer** [comment]
And then tag v1.0.1? :)

**Comment by leseb** [comment]
@alfredodeza thanks and thanks :D

**Pull request merged**